frontend/src/scenes/insights/utils/cleanFilters.ts

```diff
@@ -58,7 +58,7 @@ const cleanBreakdownParams = (
             cleanedParams['breakdown'] = filters.breakdown
         }
 
-        if (filters.breakdown_group_type_index != undefined) {
+        if (filters.breakdown_type === 'group' && typeof filters.breakdown_group_type_index === 'number') {
```

Comment:
Let's go back to `!= undefined`. Why? Because consistency - that's the way the rest of the code relating to group type indexes handles it. The original bug came _because_ this piece of code tried to reinvent the wheel.

Reply:
OK, I don't have strong feelings given one of the added tests covers someone adding an equals to remove the type juggling